build/Makefile

```diff
@@ -842,3 +842,10 @@ update-go-deps:
 	$(DOCKER_RUN) go get -u agones.dev/agones
 	$(DOCKER_RUN) go mod tidy
 	$(DOCKER_RUN) go mod vendor
+
+test-gen-crd-client:
+	mkdir -p build/tmp
+	mv ../pkg/client build/tmp
+	make gen-crd-client
+	diff -r build/tmp/client ../pkg/client || true
```

Comment:
We do want this to fail , so let's remove the || true

For example: https://github.com/googleforgames/agones/blob/02365d2e0ea3b7605824a11f4980aec1a616c153/build/includes/website.mk#L111

(Although we don't need to sort, I don't think, but if this consistently fails, let's see what other approach)

Reply:
As another though, is it worth deleting `build/tmp` if the diff passes?

Reply:
The code has been modified as per your suggestion.